.circleci/config.yml

```diff
@@ -36,6 +36,86 @@ jobs:
           prefix: coverage
       - store_test_results:
           path: test-results.xml
+  build-android:
+    docker:
+      - image: circleci/android:api-28-node
+    steps:
+      - run:
+          name: Install gradle
+          command: sudo apt install gradle
+      - run:
+          name: Install ionic
+          command: sudo npm --global install ionic
+      - run:
+          name: Install cordova
+          command: sudo npm --global install cordova@8
+      - checkout
+      - restore_cache:
+          name: Restore node modules cache
+          keys: 
+            - node-modules-{{ .Branch }}-{{ checksum "package.json" }}
+            - node-modules-{{ .Branch }}-
+            - node-modules-
+      - run:
+          name: Install node modules
+          command: npm install
+      - save_cache:
+          name: Save node modules cache
+          key: node-modules-{{ .Branch }}-{{ checksum "package.json" }}
+          paths:
+            - node_modules
+      - restore_cache:
+          name: Restoring cordova plugins cache
+          keys: 
+            - cordova-plugins-{{ .Branch }}-{{ checksum "config.xml" }}
+            - cordova-plugins-{{ .Branch }}-
+            - cordova-plugins-
+      - run:
+          name: Build android app
+          command: npx ionic cordova build android
+      - save_cache:
+          name: Save cordova plugins cache
+          key: cordova-plugins-{{ .Branch }}-{{ checksum "config.xml" }}
+          paths:
+            - plugins
+      - persist_to_workspace:
+          root: platforms/android/app/build/outputs/apk/debug/
+          paths:
+            - app-debug.apk
+  e2e-android:
+    docker:
+      - image: circleci/node:carbon
+    steps:
+      - checkout
+      - restore_cache:
+          name: Restore node modules cache
+          keys: 
+            - node-modules-{{ .Branch }}-{{ checksum "package.json" }}
+            - node-modules-{{ .Branch }}-
+            - node-modules-
+      - run:
+          name: Install node modules
+          command: npm install
+      - save_cache:
+          name: Save node modules cache
+          key: node-modules-{{ .Branch }}-{{ checksum "package.json" }}
+          paths:
+            - node_modules
+      - restore_cache:
+          name: Restoring cordova plugins cache
+          keys: 
+            - cordova-plugins-{{ .Branch }}-{{ checksum "config.xml" }}
+            - cordova-plugins-{{ .Branch }}-
+            - cordova-plugins-
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: Run E2E tests
+          command: |
+            export BROWSERSTACK_APP="$(curl -u "$BROWSERSTACK_USER:$BROWSERSTACK_KEY" \
```

Comment:
What is that service?

Reply:
In order to test and app with Browserstack we first need to upload it ( `curl [...]` ), then Browserstack   will return us and app id that we will use the run the tests so that Browserstack know against which application the tests must be executed

Reply:
Checked that yesterday and it is mega awesome.

I love the idea that we build APK and then test it, but APK will by default connect to localhost. 

If we like setup long-running sync backend we can go even further:

1. Apply config
2. Build APK
3. (Optional) Attach APK to release
4. Send APK to the Browserstack.

We can ignore IOS because it will require much more hassle to build.

Best to keep things simple now as connecting to external server may be a hassle in long run.

Reply:
There is a future in BrowserStack that create a proxy between the device and the localhost where the tests are running, so we can start the sync server (npm run start), start the Browserstack proxy and test the application using a local running sync server.

The other way is to set up a long-running OpenShift where we can deploy the sync server, but also the SSO server

Reply:
Wow really really nice. Proxy seems to be better for stability reason.